### PR TITLE
feat: Add `pause` and `isPaused` to SpriteAnimationTicker

### PR DESCRIPTION
**Pull request by munsterlander**
# Description

Based on the Discord discussion and consultation with @ufrshubham, this PR adds the ability to pause a `SpriteAnimationTicker`.

## Checklist

- [X] I have followed the [Contributor Guide] when preparing my PR.
- [X] I have updated/added tests for ALL new/updated/fixed functionality.
- [X] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [ ] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [X] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

File: packages/flame/test/sprite_animation_ticker_test.dart [hunk 1]
```diff
@@ -162,5 +162,25 @@ void main() {
...
+        [sprite, sprite],
+        stepTime: 1,
+        loop: false,
+      ).createTicker();
+
+      expect(animationTicker.isPaused, false);
+      expect(animationTicker.currentIndex, 0);
+      animationTicker.update(1);
+      expect(animationTicker.currentIndex, 1);
+      animationTicker.paused = true;
```
Comment (ufrshubham) [inline packages/flame/test/sprite_animation_ticker_test.dart:178]: Maybe do a `animationTicker.update(1)` after pausing to ensure that the `currentIndex` does not change.

**Review by ufrshubham — APPROVED** [review]
LGTM! Just a small suggestion in the test.

File: packages/flame/test/sprite_animation_ticker_test.dart (hunk 1 above)
Comment (munsterlander) [reply packages/flame/test/sprite_animation_ticker_test.dart:178]: Did I not do that?!?!  That was the whole point, lol

Comment (munsterlander) [reply packages/flame/test/sprite_animation_ticker_test.dart:178]: Added and tests pass.

**Review by spydon — APPROVED** [review]

**Pull request merged**